dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -300,28 +305,30 @@ private boolean problemFilter(Map.Entry<SymbolProblem, ClassFile> entry) {
 
       // The class is in class path but the symbol is not found
       return Optional.of(
-          new SymbolProblem(symbol, ErrorType.SYMBOL_NOT_FOUND, containingClassFile));
+          new LinkageProblem(
+              symbol, ErrorType.SYMBOL_NOT_FOUND, containingClassFile, sourceClassFile));
```

Comment:
To me it feels like the natural order is

sourceClass
targetClass
targetSymbol

and the type should be a subclass of LinkageProblem

Reply:
Updated.